src/server/debug/server/database_dump.go

```diff
@@ -0,0 +1,139 @@
+package server
+
+import (
+	"archive/tar"
+	"bufio"
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+)
+
+func (s *debugServer) collectDatabaseDump(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+	if err := s.collectDatabaseStats(ctxWithTimeout, tw, prefix...); err != nil {
+		return err
+	}
+	if err := s.collectDatabaseTables(ctxWithTimeout, tw, prefix...); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *debugServer) collectDatabaseStats(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	queries := map[string]string{
+		"table-sizes": `
+			SELECT nspname AS "schemaname", relname, pg_total_relation_size(C.oid) AS "total_size"
+			FROM pg_class C INNER JOIN pg_namespace N ON (N.oid = C.relnamespace)
+			WHERE nspname NOT IN ('pg_catalog', 'information_schema') AND C.relkind <> 'i' AND nspname !~ '^pg_toast'
+			ORDER BY nspname, relname;
+		`,
+		"activities": `
+			SELECT current_timestamp - query_start as runtime, datname, usename, client_addr, query
+			FROM pg_stat_activity WHERE state != 'idle' ORDER BY runtime DESC;
+		`,
+		"row-counts": `
+			SELECT schemaname, relname, n_live_tup, seq_scan, idx_scan
+			FROM pg_stat_user_tables ORDER BY schemaname, relname;
+		`,
+	}
+	var errs []error
+	for filename, query := range queries {
+		if err := collectDebugFile(tw, filename, "json", func(w io.Writer) error {
+			rows, err := s.database.QueryContext(ctx, query)
+			if err != nil {
+				return errors.EnsureStack(err)
+			}
+			return s.writeRowsToJSON(rows, w)
+		}, prefix...); err != nil {
+			errs = append(errs, errors.Wrapf(err, "failed to query %q", filename))
+		}
+	}
+	if len(errs) > 0 {
+		return fmt.Errorf("%v", errs)
+	}
+	return nil
+}
+
+func (s *debugServer) collectDatabaseTables(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	tables, err := pachsql.ListTables(ctx, s.database)
+	if err != nil {
+		return errors.EnsureStack(err)
+	}
+	for _, table := range tables {
+		fullPrefix := strings.Join(append(append([]string{}, prefix...), "tables", table.SchemaName), "/")
+		if err := collectDebugFile(tw, table.TableName, "json", func(w io.Writer) error {
+			bufferedWriter := bufio.NewWriter(w)
+			if err := s.collectTable(ctx, bufferedWriter, &table); err != nil {
+				return errors.Wrapf(err, "error collecting table %s.%s", table.SchemaName, table.TableName)
```

Comment:
There are a bunch of these so I'll mention it one more time, but this can be `collect table %s.%s`; we know it's an error because it's of type `error`.